### PR TITLE
Add goverter:mapExtend

### PR DESCRIPTION
**Pull request by soniah**
Hi Jannis, here's a start - parsing. Comments?

I think the name `mapExtend` is fine, as `Extend` lines up with the "external" `// goverter:extend`, and it's a mapping. You could call it `mapFunction` but then you've got `Extend` verses `Function`. 

Also I added in a feature for limiting the tests that are run, I could remove this later if you don't like it. I'd remove the `spew` module before release.

### CODE REVIEW

File: runner_test.go [hunk 1]
```diff
@@ -25,7 +26,7 @@ func TestScenario(t *testing.T) {
 	require.NoError(t, os.MkdirAll(execDir, 0755))
 	require.NoError(t, clearDir(execDir))
 
-	for _, file := range files {
+	for _, file := range filteredFiles(t, files) {
```
Comment (jmattheis) [inline runner_test.go:29]: Something like this is already built into `go test`, use the `-run` flag.
```bash
$ go test ./ -run TestScenario/3_struct_mapping_extend.yml
--- FAIL: TestScenario (0.04s)
    --- FAIL: TestScenario/3_struct_mapping_extend.yml (0.04s)
         ....
```

File: scenario/3_struct_mapping_extend.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+input:
```
Comment (jmattheis) [inline scenario/3_struct_mapping_extend.yml:1]: Future test cases this feature needs:  (the testcases should fail with an appropriate error message)
- `goverter:mapExtends` with a function reference that doesn't exist
- `goverter:mapExtends` with a non function reference. F.ex `DefaultAge` is a constant. (this should fail)
- `goverter:mapExtends` with a function reference with parameters
- `goverter:mapExtends` with a function reference without return type
- `goverter:mapExtends` with a function reference with wrong return type. Like the output struct requires string but the method return int
- `goverter:mapExtends` with a function reference with multiple return values

File: scenario/3_struct_mapping_extend.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    input.go: |
+        package structs
```
Comment (jmattheis) [inline scenario/3_struct_mapping_extend.yml:3]: Is your plan to incrementally update this PR with all the features?

**Review by jmattheis — CHANGES_REQUESTED** [review]
Hey, thanks for the PR. I've added sub comments with remarks.

File: runner_test.go (hunk 1 above)
Comment (soniah) [reply runner_test.go:29]: Duh! I just need to configure my editor (Goland) correctly.

Reverted.

File: scenario/3_struct_mapping_extend.yml (hunk 3 above)
Comment (soniah) [reply scenario/3_struct_mapping_extend.yml:3]: Yes. I wanted to sketch out the design first, see if we're on the same page 😄

**Comment by leedsalex** [comment]
@soniah @jmattheis Any update on this PR, I could really use this feature.

**Comment by soniah** [comment]
No. I lost everything in the recent floods in Australia, I've been worried
about things like food, clothes and shelter 😢

https://www.theguardian.com/australia-news/video/2022/mar/31/nsw-floods-drone-footage-shows-scale-of-devastation-in-lismore-video

**Comment by waltcow** [comment]
> @soniah @jmattheis Any update on this PR, I could really use this feature.

https://github.com/waltcow/goverter/tree/feat_map_extend

**Pull request closed without merging**